### ISSUE TITLE
Union freshness is not removed when required

### ISSUE THREAD

**Issue by weswigham** [Microsoft/TypeScript#19226]
**TypeScript Version:**  2.6.0-dev.201xxxxx

**Code**

```ts
import * as React from "react";
import { SFC } from "react";


type CommonProps = { common: string };
type Props = CommonProps &
    ({ type: 'foo' } | { type: 'bar'; message: string });

const Button: SFC<Props> = (props) => null;

const OtherButton1: SFC<Props> = (props) => {
    const props2 = {...props};
    return React.createElement(Button, props2); // `props2` should be fine, just like `props` is
};
```

**Expected behavior:**
No errors

**Actual behavior:**
```
test.tsx(13,40): error TS2345: Argument of type '{ common: string; type: "foo"; children?: ReactNode; } | { common: string; type: "bar"; message: ...'
is not assignable to parameter of type '(Attributes & CommonProps & { type: "foo"; }) | (Attributes & CommonProps & { type: "bar"; messag...'.
  Type '{ common: string; type: "foo"; children?: ReactNode; }' is not assignable to type '(Attributes & CommonProps & { type: "foo"; }) | (Attributes
& CommonProps & { type: "bar"; messag...'.
    Object literal may only specify known properties, and 'children' does not exist in type 'Attributes & CommonProps & { type: "foo"; }'.
```

Found when revisiting some test code for something else.

cc @mhegazy @sandersn

**Comment by sandersn** [comment]
Here's a standalone repro:

```ts
declare let t0: { a, b } | { d, e }
declare let t1: { a, b, c } | { c, d, e }
t0 = t1 // OK
t0 = { ...t1 } // error: 'c' is fresh

let tmp = { ...t1 }
t0 = tmp // should be OK, but is currently "error: 'c' is fresh"
```

Assignment to `tmp` should get rid of freshness from the spread type, but doesn't.

**Comment by sandersn** [comment]
`getSpreadType` no longer added `TypeFlags.ContainsObjectLiteral` after I moved the flag-setting code there.

**Comment by sandersn** [comment]
Fix is up at #19250
